Adding cookie banner

mavi-sqr opened the pull request:
As part of the Nextflow.io redesign, I decided to align this repo with the rest of the product ecosystem by introducing React and TailwindCSS. The goal is to:

- Ensure consistency across all Nextflow products.
- Leverage React’s component-based structure for better state management and interactivity.
- Improve styling scalability with Tailwind, reducing custom CSS overhead.

Changes Introduced

- Added React (@astrojs/react) and TailwindCSS (@astrojs/tailwind) to the project.
- Refactored the project to use React where applicable, aligning with the stack used in other products.
- Implemented a Cookie Banner that:
  - Uses react-cookie to store user preferences.
  - Integrates with Google Analytics (gtag) and PostHog for tracking consent.
  
  
  SUMMARY

>   1. I deleted podcast and blog content as we redirect it right now
> 2. Added Cookie banner
> 3. Deleted blog and podcast content and pages.
> 4. Also deleted tag page.
> 5. I needed to fix some UI bugs. Adding Tailwind caused some styles to overwrite while I continued using Bootstrap. Please take extra care when reviewing this, as I may have overlooked some styles. :face_exhaling:

Next Steps

- [x] A full QA session is planned for tomorrow to verify cookie storage, tracking consent, and component behavior across different environments.
- [x] Open for feedback and review on the React & Tailwind adoption before proceeding with further refactoring.

mavi-sqr commented on public/css/custom.css:
```diff
@@ -197,6 +197,9 @@ img {
   .navbar-nav > li > a {
     padding: 22.5px 10px;
   }
+  .navbar-collapse.collapse {
```
This is due to Bootstram Tailwind conflicts. Will be erased.

ewels commented:
Font weights look off for me in the preview:

![CleanShot 2025-03-17 at 14 09 14@2x](https://github.com/user-attachments/assets/49abb1f8-99c3-4ecf-b04d-49665373dac0)

ewels commented:
Text alignment on the homepage looks a bit wonky. On the old site the columns are narrower so it's less obvious. Maybe worth adding a `text-align: justify` on it?
![CleanShot 2025-03-17 at 14 23 04@2x](https://github.com/user-attachments/assets/98986980-2c25-4091-b40a-8bf2373cb71c)

I can't click the cookie banner away, clicking seems to have no effect for me.

That's all from my quick first-pass visual only 👍🏻 

There is a bunch of veeery outdated content there 🫠 Maybe I can put in another PR after this is merged to solve that though, so I don't cause a bunch of merge conflicts.

dana-seqera commented:
About page
- padding beneath horizontal lines
- side padding on contributors

 <img width="2183" alt="Screenshot 2025-03-17 at 10 25 02 AM" src="https://github.com/user-attachments/assets/237ea21b-dedc-46dc-b2c9-c37262eb04a9" />

<img width="2499" alt="Screenshot 2025-03-17 at 10 25 16 AM" src="https://github.com/user-attachments/assets/6730ae6f-87eb-4030-8061-a521ec94b810" />

dana-seqera commented:
Line break after Java on homepage "Check prerequisites" block

<img width="788" alt="Screenshot 2025-03-17 at 10 26 15 AM" src="https://github.com/user-attachments/assets/38203b3a-e810-4308-9dc8-0b2ddea0ed99" />

mavi-sqr commented:
Thanks, both of you @dana-seqera @ewels 
- I switched to inter to be one step forward. But happy to go back to Arial.
- Same with code font family. I missed that one, we are going back to Menlo.
It's weird that the cookie banner buttons were not working. I changed client:load to` client:only="react"`, the component will not exist in the initial HTML; it will only appear when React mounts it on the client.

dana-seqera reviewed (approved):
LGTM! 👍
